src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectWorkspaceStateGenerator.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    [Shared]
+    [Export(typeof(ProjectWorkspaceStateGenerator))]
+    [Export(typeof(ProjectSnapshotChangeTrigger))]
+    internal class ProjectWorkspaceStateGenerator : ProjectSnapshotChangeTrigger
```

Comment:
Let's talk about this, it's not clear to me why we need this. `WorkspaceProjectSnapshotChangeTrigger` already implements a backoff timer